internal/dashboard/root.go

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright © 2023 NAME HERE cosmo-workspace
+*/
+package dashboard
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"reflect"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	klog "k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	cosmov1alpha1 "github.com/cosmo-workspace/cosmo/api/v1alpha1"
+	"github.com/cosmo-workspace/cosmo/pkg/auth"
+	"github.com/cosmo-workspace/cosmo/pkg/clog"
+	"github.com/cosmo-workspace/cosmo/pkg/kosmo"
+)
+
+const (
+	fieldManager string = "cosmo-dashboard"
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(cosmov1alpha1.AddToScheme(scheme))
+	// +kubebuilder:scaffold:scheme
+}
+
+type options struct {
+	KubeConfigPath string
+	KubeContext    string
+
+	ZapOpts zap.Options
+
+	Logr *clog.Logger
+	// Client *kosmo.Client
+
+	StaticFileDir           string
+	CookieDomain            string
+	CookieHashKey           string
+	CookieBlockKey          string
+	CookieSessionName       string
+	ResponseTimeoutSeconds  int64
+	GracefulShutdownSeconds int64
+	TLSPrivateKeyPath       string
+	TLSCertPath             string
+	Insecure                bool
+	ServerPort              int
+	MaxAgeMinutes           int
+}
+
+func NewRootCmd(o *options) *cobra.Command {
+
+	rootCmd := &cobra.Command{
+		Use:   "authserver",
+		Short: "Authenticate server",
+		Long: `
+cosmo dashboard server
+Complete documentation is available at http://github.com/cosmo-workspace/cosmo
+
+MIT 2023 cosmo-workspace/cosmo
+`,
+		Version:           "v0.8.0 cosmo-workspace 2023",
+		PersistentPreRunE: o.PreRunE,
+		RunE:              o.RunE,
+	}
+
+	goflags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	klog.InitFlags(goflags)
+	o.ZapOpts.BindFlags(goflags)
+	ctrl.RegisterFlags(goflags)
+	rootCmd.Flags().AddGoFlagSet(goflags)
+
+	rootCmd.PersistentFlags().Int64Var(&o.ResponseTimeoutSeconds, "timeout-seconds", 3, "Timeout seconds for response")
+	rootCmd.PersistentFlags().Int64Var(&o.GracefulShutdownSeconds, "graceful-shutdown-seconds", 10, "Graceful shutdown seconds")
+	rootCmd.PersistentFlags().StringVar(&o.StaticFileDir, "serve-dir", "/app/public", "Static file dir to serve")
+	rootCmd.PersistentFlags().StringVar(&o.CookieDomain, "cookie-domain", "", "Cookie domain name")
+	rootCmd.PersistentFlags().StringVar(&o.CookieHashKey, "cookie-hashkey", "", "Cookie hashkey")
```

Comment:
TODO: better that sensitive values are given by secret env

Reply:
Give these values in secret env.
Is there another way?
https://github.com/cosmo-workspace/cosmo/blob/main/charts/cosmo-traefik/templates/cosmo-auth-middleware.yaml